src/test/ui/traits/trait-upcasting/unsafe-trait-upcasting-coercion2.rs

```diff
@@ -0,0 +1,29 @@
+#![feature(trait_upcasting)]
+#![allow(incomplete_features)]
+
+trait A {
+    fn foo_a(&self) {}
+}
+
+trait B {
+    fn foo_b(&self) {}
+}
+
+trait C: A + B {
+    fn foo_c(&self) {}
+}
+
+struct S;
+
+impl A for S {}
+
+impl B for S {}
+
+impl C for S {}
+
+fn invalid_coercion(v: * const Box<dyn C>) {
+    let g: * const Box<dyn A> = v;
+    //~^ ERROR mismatched types
+}
+
+fn main() {}
```

Comment:
Why are these two separate files?

Please also add some casts that *are* okay.

Does this hit the last match arm in `is_trait_upcasting_coercion_from_raw_pointer`?

Reply:
This is a separate file because the "mismatched types" error will prevent unsafety checker from running at all, it seems. I don't really know whether error recovery is possible at all in this case. This test case is just the status quo. I'll leave comments, when the unsafety checking implementation is ready.

> Please also add some casts that are okay.
I will, thanks!

> Does this hit the last match arm in is_trait_upcasting_coercion_from_raw_pointer?

No it doesn't, the unsafety checker is not yet running in this case.

Reply:
> No it doesn't, the unsafety checker is not yet running in this case.

Okay, then the other test should definitely do something to hit that arm.